src/IO/Exporter/Exporter.cpp

```diff
@@ -165,21 +165,29 @@ std::vector<std::vector<double>> interpolate_to_points(
     ERROR_NO_TRACE("Mismatched dimensions: expected "
                    << Dim << "D volume data, but got " << dim << "D.");
   }
-  // Get observation ID. This currently assumes that all volume files contain
-  // the same observations. For generalizing to volume files across multiple
-  // segments, see the Python function `Visualization.ReadH5:select_observation`
-  // and possibly move it to C++.
-  const auto obs_ids = first_volfile.list_observation_ids();
-  if (observation_step < 0) {
-    observation_step += static_cast<int>(obs_ids.size());
-  }
-  if (observation_step < 0 or
-      static_cast<size_t>(observation_step) >= obs_ids.size()) {
-    ERROR_NO_TRACE("Invalid observation step: "
-                   << observation_step << ". There are " << obs_ids.size()
-                   << " observations in the file.");
-  }
-  const size_t obs_id = obs_ids[static_cast<size_t>(observation_step)];
+  // Get observation ID
+  const size_t obs_id = std::visit(
```

Comment:
This feels like it should have a test. Is that easy to add?

Reply:
Would this be easy to add? Also, do we need to worry about automatic conversion between `size_t` and `int`? Currently I think it's only tested for `int`? Actually I'm not sure because of the conversion thing.